public/app/plugins/panel/table2/module.tsx

```diff
@@ -1,39 +1,44 @@
-import { PanelPlugin } from '@grafana/data';
+import { PanelPlugin, FieldConfigProperty } from '@grafana/data';
 import { TablePanel } from './TablePanel';
 import { CustomFieldConfig, defaults, Options } from './types';
 
 export const plugin = new PanelPlugin<Options, CustomFieldConfig>(TablePanel)
   .setDefaults(defaults)
-  .setCustomFieldOptions(builder => {
-    builder
-      .addNumberInput({
-        id: 'width',
-        name: 'Column width',
-        description: 'column width (for table)',
-        settings: {
-          placeholder: 'auto',
-          min: 20,
-          max: 300,
-        },
-        defaultValue: 1,
-      })
-      .addSelect({
-        id: 'displayMode',
-        name: 'Cell display mode',
-        description: 'Color value, background, show as gauge, etc',
-        settings: {
-          options: [
-            { value: 'auto', label: 'Auto' },
-            { value: 'color-background', label: 'Color background' },
-            { value: 'gradient-gauge', label: 'Gradient gauge' },
-            { value: 'lcd-gauge', label: 'LCD gauge' },
-          ],
-        },
-      });
+  .useFieldConfigOptions({
+    standardOptions: [FieldConfigProperty.Min],
+    standardOptionsDefaults: {
+      [FieldConfigProperty.Min]: 10,
+    },
+    useCustomOptions: builder => {
+      builder
+        .addNumberInput({
+          path: 'width',
+          name: 'Column width',
+          description: 'column width (for table)',
+          settings: {
+            placeholder: 'auto',
+            min: 20,
+            max: 300,
+          },
+        })
+        .addSelect({
+          path: 'displayMode',
+          name: 'Cell display mode',
+          description: 'Color value, background, show as gauge, etc',
+          settings: {
+            options: [
+              { value: 'auto', label: 'Auto' },
+              { value: 'color-background', label: 'Color background' },
+              { value: 'gradient-gauge', label: 'Gradient gauge' },
+              { value: 'lcd-gauge', label: 'LCD gauge' },
+            ],
+          },
+        });
+    },
   })
```

Comment:
This is the main new api to set custom field options, and optionally set order & defaults and what options to show 
@ryantxu

Reply:
all properties are optional in this call so if you do not pass anything of if any property is not specified defaults will be used